### ISSUE TITLE
Nest directories in --ignore-dirs

### ISSUE THREAD

**Issue by ruoti** [depcheck/depcheck#138]
If I put a nested directory for --ignore-dirs (e.g., --ignore-dirs=src/resources), the directory is not ignored.

Also, on the README.md, --ignore-dirs is written as --ignores-dirs once.

**Comment by lijunle** [comment]
Hi, @mathflair Thanks for reporting issue.

1. The nest directories should be supported. I will take a look.
2. The README typo will be fixed soon.

**Comment by KrekkieD** [comment]
I have the same issue. Reason is found in `check.js`, in the `ignoreDirs.indexOf` part.

The `_path2.default.basename(subdir)` will truncate the provided `subdir` to the last folder name. That means when it's checking for `src/resources` to be ignored, it truncates it to `resources` and uses that value to search in the `--ignore-dirs` values.

I suppose the solution would be to use `_path.relative` instead:

```
      return ignoreDirs.indexOf(_path.relative(rootDir, subdir)) === -1 && !isModule(subdir) ? promises.push(checkDirectory(subdir, rootDir, ignoreDirs, deps, parsers, detectors)) : null;
```

**Comment by pdehaan** [comment]
I too have this problem, where my sad `--ignore-dirs=` value is a horrible, deeply nested:

```sh
--ignore-dirs=data,test/fixtures/mock-sdk/lib/sdk/addon
```

**Comment by goloroden** [comment]
Any update on this?

**Comment by rumpl** [comment]
None yet, gotta take a look a it

**Comment by znarf** [comment]
`ignore-dirs` is only a blacklist of single level directory name, it doesn't support nesting.

If you want to skip `test/fixtures/mock-sdk/lib/sdk/addon` you need to try to skip `test`, `fixtures`, `mock-sdk`, `lib`, `sdk` or `addon`. Not great!

I'm attempting to fix that as part of https://github.com/depcheck/depcheck/pull/498
